internal/mode/static/policies/clientsettings/validator.go

```diff
@@ -0,0 +1,215 @@
+			fmt.Sprintf("unsupported targetRef Group %q; must be %s", ref.Group, gatewayv1.GroupName),
+		)
+	}
+
+	supportedKinds := []gatewayv1.Kind{kinds.Gateway, kinds.HTTPRoute, kinds.GRPCRoute}
+
+	if !slices.Contains(supportedKinds, ref.Kind) {
+		path := basePath.Child("kind")
+
+		return field.Invalid(
+			path,
+			ref.Kind,
+			fmt.Sprintf("unsupported targetRef Kind %q; Kind must be one of: %v", ref.Kind, supportedKinds),
+		)
+	}
+
+	return nil
+}
+
+func (v *Validator) validateSettings(spec ngfAPI.ClientSettingsPolicySpec) error {
+	var allErrs field.ErrorList
+	fieldPath := field.NewPath("spec")
+
+	if spec.Body != nil {
+		allErrs = append(allErrs, v.validateClientBody(*spec.Body, fieldPath.Child("body"))...)
+	}
+
+	if spec.KeepAlive != nil {
+		allErrs = append(allErrs, v.validateClientKeepAlive(*spec.KeepAlive, fieldPath.Child("keepAlive"))...)
+	}
+
+	return allErrs.ToAggregate()
+}
+
+func (v *Validator) validateClientBody(body ngfAPI.ClientBody, fieldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if body.Timeout != nil {
+		if err := v.genericValidator.ValidateNginxDuration(string(*body.Timeout)); err != nil {
+			path := fieldPath.Child("timeout")
+
+			allErrs = append(allErrs, field.Invalid(path, body.Timeout, err.Error()))
+		}
+	}
+
+	if body.MaxSize != nil {
+		if err := v.genericValidator.ValidateNginxSize(string(*body.MaxSize)); err != nil {
+			path := fieldPath.Child("maxSize")
+
+			allErrs = append(allErrs, field.Invalid(path, body.MaxSize, err.Error()))
+		}
+	}
+
+	return allErrs
+}
+
+func (v *Validator) validateClientKeepAlive(keepAlive ngfAPI.ClientKeepAlive, fieldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if keepAlive.Requests != nil {
+		requests := *keepAlive.Requests
+		if requests < 0 {
+			path := fieldPath.Child("requests")
+
+			allErrs = append(
+				allErrs,
+				field.Invalid(path, *keepAlive.Requests, "requests is invalid: must be positive"),
+			)
+		}
+	}
+
+	if keepAlive.Time != nil {
+		if err := v.genericValidator.ValidateNginxDuration(string(*keepAlive.Time)); err != nil {
+			path := fieldPath.Child("time")
+
+			allErrs = append(allErrs, field.Invalid(path, *keepAlive.Time, err.Error()))
+		}
+	}
+
+	if keepAlive.Timeout != nil {
+		timeout := keepAlive.Timeout
+
+		if timeout.Server != nil {
+			if err := v.genericValidator.ValidateNginxDuration(string(*timeout.Server)); err != nil {
+				path := fieldPath.Child("timeout").Child("server")
+
+				allErrs = append(
+					allErrs,
+					field.Invalid(path, *keepAlive.Timeout.Server, err.Error()),
+				)
+			}
+		}
+
+		if timeout.Header != nil {
+			if err := v.genericValidator.ValidateNginxDuration(string(*timeout.Header)); err != nil {
+				path := fieldPath.Child("timeout").Child("header")
+
+				allErrs = append(
+					allErrs,
+					field.Invalid(path, *keepAlive.Timeout.Header, err.Error()),
+				)
+			}
+		}
+
+		if keepAlive.Timeout.Header != nil && keepAlive.Timeout.Server == nil {
```

Comment:
since we are using the same validation for all these fields, maybe there is a way to refactor this?

Reply:
That's a good suggestion, but I don't see a way to refactor this that would reduce the number of lines of code.